include/unifex/sync_wait.hpp

```diff
@@ -125,27 +111,18 @@ using receiver_t = typename _receiver<T>::type;
 
 template <typename Result, typename Sender>
 std::optional<Result> _impl(Sender&& sender) {
-  auto blockingResult = blocking(sender);
-  const bool completesSynchronously =
-      blockingResult == blocking_kind::always ||
-      blockingResult == blocking_kind::always_inline;
-
   using promise_t = _sync_wait::promise<Result>;
   promise_t promise;
-  if (!completesSynchronously) {
-    promise.doneEvent_.emplace();
-  }
+  manual_event_loop ctx;
 
   // Store state for the operation on the stack.
   auto operation = connect(
-      ((Sender &&) sender),
-      _sync_wait::receiver_t<Result>{promise});
+      (Sender&&)sender,
+      _sync_wait::receiver_t<Result>{promise, ctx});
```

Comment:
We've lost the optimisation here that efficiently handles senders that complete synchronously without synchronisation.

Is there a way we can retain this while still allowing the forward-progress delegation properties of this implementation?

Reply:
If it is *guaranteed* to complete inline, then constexpr if checking that should be safe. It has to be guaranteed, though.

Reply:
I don't think we have the mechanism to query for 'blocking()` at compile-time yet.
I need to add some helper classes for supporting this which were mentioned in the tag_invoke paper.